raster/src/main/scala/geotrellis/raster/io/geotiff/MultibandGeoTiff.scala

```diff
@@ -136,6 +129,32 @@ case class MultibandGeoTiff(
     MultibandGeoTiff(overviewTile, overviewRasterExtent.extent, crs, Tags.empty, overviewOptions)
   }
 
+  def withOverviews(resampleMethod: ResampleMethod, decimations: List[Int] = Nil, blockSize: Int = GeoTiff.DefaultBlockSize): MultibandGeoTiff = {
+    val overviewDecimations: List[Int] =
+      if (decimations.isEmpty) {
+        GeoTiff.defaultOverviewDecimations(tile.cols, tile.rows, blockSize)
+      } else {
+        decimations
+      }
+
+    if (overviewDecimations.isEmpty) {
+      this
+    } else {
+      // force ArrayTile to avoid costly compressor thrashing in GeoTiff segments when resample will stride segments
+      val arrayTile = tile match {
+        case tiffTile: GeoTiffMultibandTile =>
+          tiffTile.toArrayTile() // allow GeoTiff tile to read segments in optimal way
+        case _ =>
+          MultibandTile(tile.bands.map(_.toArrayTile()))
+      }
+      val staged = MultibandGeoTiff(arrayTile, extent, crs, tags, options, Nil)
+      val overviews = overviewDecimations.map { (decimationFactor: Int) =>
+        staged.buildOverview(resampleMethod, decimationFactor, blockSize)
+      }
+      MultibandGeoTiff(tile, extent, crs, tags, options, overviews)
```

Comment:
@pomadchin What do you think is more appropriate here?
Should we return the base tile, which could be lazy or compressed, or to return the forced `ArrayTile` since we already went through the trouble of fetching it.

Reply:
@echeipesh from the API perspective, I'd expect to see the result of function as a geotiff. What's your point for the `ArrayTile`?

Reply:
In order to generate the overviews in a way that match the base tile resample and save on segment thrashing the overview code will pull the base GeoTiff into `ArrayTile` and resample that.

The question that I have: given that we do that, should the returned `GeoTiff` now hold the `ArrayTile` version of the segments or the original geotiff segments, which could be lazy. I was thinking the latter, but wanted a double check.

Reply:
Yea, `the original geotiff segments` sounds correct :+1: